iceoryx_binding_c/test/moduletests/test_subscriber.cpp

```diff
@@ -245,3 +259,52 @@ TEST_F(iox_sub_test, sendingTooMuchLeadsToLostChunks)
 
     EXPECT_TRUE(iox_sub_has_lost_chunks(m_sut));
 }
+
+TEST_F(iox_sub_test, attachingToWaitSetWorks)
+{
+    EXPECT_EQ(iox_sub_attach_to_ws(m_sut, m_waitSet, SubscriberEvent_HAS_NEW_SAMPLES, 0, NULL), WaitSetResult_SUCCESS);
+    EXPECT_EQ(m_waitSet->size(), 1);
+}
+
+TEST_F(iox_sub_test, detachingFromWaitSetWorks)
+{
+    iox_sub_attach_to_ws(m_sut, m_waitSet, SubscriberEvent_HAS_NEW_SAMPLES, 0, NULL);
+    iox_sub_detach_ws(m_sut);
+    EXPECT_EQ(m_waitSet->size(), 0);
+}
+
+TEST_F(iox_sub_test, hasNewSamplesTriggersWaitSetWithCorrectTriggerId)
+{
+    iox_sub_attach_to_ws(m_sut, m_waitSet, SubscriberEvent_HAS_NEW_SAMPLES, 587, NULL);
+
```

Comment:
I would prefer having the empty lines only between the AAA sections. Like this it's harder to identify them

Reply:
done